mlrun/api/api/utils.py

```diff
@@ -150,12 +156,169 @@ async def submit_run(db_session: Session, auth_info: mlrun.api.schemas.AuthInfo,
+    _obfuscate_v3io_access_key_env_var(function, auth_info)
+    _obfuscate_v3io_volume_credentials(function)
+
+
+def _obfuscate_v3io_volume_credentials(function):
+    if (
+        function.kind
+        and function.kind not in mlrun.runtimes.RuntimeKinds.local_runtimes()
+    ):
+        get_item_attribute = mlrun.runtimes.utils.get_item_name
+        function: mlrun.runtimes.pod.KubeResource
+        v3io_volume_indices = []
+        for index, volume in enumerate(function.spec.volumes):
+            flex_volume = get_item_attribute(
+                volume, "flexVolume"
+            ) or get_item_attribute(volume, "flex_volume")
+            if flex_volume and get_item_attribute(flex_volume, "driver") == "v3io/fuse":
+                v3io_volume_indices.append(index)
+        # if it's already a V1EnvVarSource instance, it's already been obfuscated
+        if v3io_volume_indices:
+            volume_name_to_volume_mounts = collections.defaultdict(list)
+            for volume_mount in function.spec.volume_mounts:
+                # sanity
+                if not get_item_attribute(volume_mount, "name"):
+                    logger.warning(
+                        "Found volume mount without name, skipping obfuscation",
+                        volume_mount=volume_mount,
+                    )
+                    continue
+                volume_name_to_volume_mounts[
+                    get_item_attribute(volume_mount, "name")
+                ].append(volume_mount)
+            for index in v3io_volume_indices:
+                volume = function.spec.volumes[index]
+                flex_volume = get_item_attribute(
+                    volume, "flexVolume"
+                ) or get_item_attribute(volume, "flex_volume")
+                if isinstance(flex_volume, kubernetes.client.V1FlexVolumeSource):
+                    flex_volume = flex_volume.to_dict()
+                # if it's already referencing a secret, nothing to do
+                if flex_volume.get("secretRef"):
+                    continue
+                access_key = flex_volume.get("options", {}).get(
+                    "accessKey"
+                ) or flex_volume.get("options", {}).get("access_key")
+                # sanity
+                if not access_key:
+                    logger.warning(
+                        "Found v3io fuse volume without access key, skipping obfuscation",
+                        volume=volume,
+                    )
+                    continue
+                if not get_item_attribute(volume, "name"):
+                    logger.warning(
+                        "Found volume without name, skipping obfuscation", volume=volume
+                    )
+                    continue
+                volume_name = get_item_attribute(volume, "name")
+                # Best effort - usually we'll have a volume mount to /users/<username>, try to resolve from there
+                # if not found, skip obfuscation
```

Comment:
Discussed offline with @theSaarco and @Tankilevitch and decided that:
1. For `MLRUN_AUTH_SESSION`:
*  Use the username from `auth_info`, if it's not there explode.
* We are aware it may not be the right username (someone my give other user's access key in `function.metadata.credentials.access_key`), but nothing is really using the username in the secret for this matter. it will make revocation a bit more complex, we're ok with it
2. For `V3IO_ACCESS_KEY`:
* Use the `V3IO_USERNAME` env var, if it's not there, fallback to `auth_info`, if it's not there, explode
* Again, this may not be the actually matching username, but it's ok since it's not used
* usually (99%) it will be set automatically through our auto mount which uses our `mlrun.v3io_cred` which sets the both the access key and the username env var, therefore we're looking on it first  
3. For the v3io fuse volume:
* Use the username resolved from the mount, if not found, use the `V3IO_USERNAME`, if it's not there, skip obfuscation
* Since the username field of the secret is used by the fuse volume, we prefer to prevent having username that doesn't match the access key over doing obfuscation.
* v3io fuse volume is usually done through our `mlrun.mount_v3io` which by default also calls `mlrun.v3io_cred`, therefore we're assuming `V3IO_USERNAME` is the relevant username